addons/binding/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/handler/AstroThingHandler.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.astro.handler;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.astro.internal.config.AstroThingConfig;
+import org.openhab.binding.astro.internal.job.JobScheduler;
+import org.openhab.binding.astro.internal.model.Planet;
+import org.openhab.binding.astro.internal.util.PropertyUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base ThingHandler for all Astro handlers.
+ * 
+ * @author Gerhard Riegler - Initial contribution
+ */
+public abstract class AstroThingHandler extends EqualStateFilterHandler {
+    private static final Logger logger = LoggerFactory.getLogger(AstroThingHandler.class);
+    private JobScheduler jobScheduler = new JobScheduler();
+    protected AstroThingConfig thingConfig;
+
+    public AstroThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        String thingUid = getThing().getUID().toString();
+        thingConfig = getConfigAs(AstroThingConfig.class);
+        thingConfig.setThingUid(thingUid);
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(thingConfig.getGeolocation()) == null) {
+            logger.error("Astro parameter geolocation is mandatory and must be configured, disabling thing '{}'", thingUid);
+            validConfig = false;
+        } else {
+            thingConfig.parseGeoLocation();
+        }
+
+        if (thingConfig.getLatitude() == null || thingConfig.getLongitude() == null) {
+            logger.error("Astro parameters geolocation could not be split into latitude and longitude, disabling thing '{}'", thingUid);
+            validConfig = false;
+        }
+        if (thingConfig.getInterval() == null || thingConfig.getInterval() < 1 || thingConfig.getInterval() > 86400) {
+            logger.error("Astro parameter interval must be in the range of 1-86400, disabling thing '{}'", thingUid);
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            logger.debug(thingConfig.toString());
+            updateStatus(ThingStatus.ONLINE);
+            jobScheduler.start(thingConfig);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        jobScheduler.stop(thingConfig);
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
This is already done by the framework and must be removed here as it leads to an exception.